packages/remix-react/components.tsx

```diff
@@ -929,16 +1171,177 @@ export function useFetcher<TData = any>(): FetcherWithComponents<
   SerializeFrom<TData>
 > {
   let fetcherRR = useFetcherRR();
-  let { Form, submit, load } = fetcherRR;
-
-  // TODO: For SSR we can just return IDLE_FETCHER for now, we'll handle
-  // this as part of hydration work
+  let remixFetcher = convertRouterFetcherToRemixFetcher({
+    state: fetcherRR.state,
+    data: fetcherRR.data,
+    formMethod: fetcherRR.formMethod,
+    formAction: fetcherRR.formAction,
+    formData: fetcherRR.formData,
+    formEncType: fetcherRR.formEncType,
+    " _hasFetcherDoneAnything ": fetcherRR[" _hasFetcherDoneAnything "],
+  });
   return {
-    Form,
-    submit,
-    load,
-    ...IDLE_FETCHER,
+    ...remixFetcher,
+    load: fetcherRR.load,
+    submit: fetcherRR.submit,
+    Form: fetcherRR.Form,
+  };
+}
+
+function convertRouterFetcherToRemixFetcher(
+  fetcherRR: Omit<ReturnType<typeof useFetcherRR>, "load" | "submit" | "Form">
+): Fetcher {
+  let { state, formMethod, formAction, formEncType, formData, data } =
+    fetcherRR;
+
+  let isActionSubmission =
+    formMethod != null &&
+    ["POST", "PUT", "PATCH", "DELETE"].includes(formMethod.toUpperCase());
+
+  if (state === "idle") {
+    if (fetcherRR[" _hasFetcherDoneAnything "] === true) {
```

Comment:
Needed _something_ here to differentiate fetcher `idle`/`init` and `idle`/`done`

Reply:
For the `useFetcher` use-case we could do this by tracking some local state (`let didAnything = useState(false)`).  

However, in `useFetchers` we don't have a stable key to track fetchers so we cant do the same there.  

In user application code, they can also derive this "done" state via their own `useState` if needed.